src/registry-buddy/image/dynamic_deleter.go

```diff
@@ -0,0 +1,25 @@
+package image
+
+import (
+	"code.cloudfoundry.org/capi-k8s-release/src/registry-buddy/dockerhub"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"log"
+	"strings"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 -o fakes/deleter.go --fake-name Deleter . Deleter
+type Deleter func(name.Reference, authn.Authenticator, *log.Logger) error
+
+func NewDynamicDeleter() Deleter {
```

Comment:
For better or for worse, I like that this sets us up to handle these special cases easier going forward. 🙂

Reply:
I was initially hoping that we could use a factory when the app started and just have a single static deletion function. Unfortunately, though, we don't know which registry URL we're using until we get a delete request. That required me to create this "dynamic" deleter which decides for each request which deleter to use.